README.md

```diff
@@ -1 +1,196 @@
+
+## 2.2 Raw (Noncanonicalized) Data
+
+Canonicalization is not required if data is encoded as raw bytes (multicodec `0x00`). The exact bytes are already present, and MUST not be changed.
+
+# 3 Varsig Format
+
+A varsig is a bytestring that includes the following information:
+
+| Segment Name              | Type     | Description                                     | Required |
+|---------------------------|----------|-------------------------------------------------|----------|
+| Varsig Prefix             | `0xD000` | The multicodec varsig prefix                    | Yes      |
+| Public Key Prefix         | `Varint` | The multicodec prefix for the public key type   | Yes      |
+| Hash Prefix               | `Varint` | The multicodec prefix for the hash algorithm    | Yes      |
+| Hash Length               | `Varint` | The hash length                                 | Yes      |
+| Content Multicodec Prefix | `Varint` | The IPLD encoding uses to canonicalize the data | Yes      |
+| Raw Signature             | `Varint` | The raw signature                               | Yes      |
+  
+## 3.1 Segments
+
+### 3.1.1 Varsig Prefix
+
+The varsig prefix MUST be `0xD000`.
+
+### 3.1.2 Public Key Prefix
+
+The [multicodec](https://github.com/multiformats/multicodec) prefix for the public key type associated with the signature.
+
+For example: `0xE7` for secp256k1, or `0x1205` for RSA.
+
+### 3.1.3 Hash Prefix
+
+The [multicodec](https://github.com/multiformats/multicodec) hash prefix used by the signature. This is the first segment of a [multihash](https://github.com/multiformats/multihash).
+
+For example: `0x16` for SHA3-256, or `0x1E` for BLAKE3
+
+### 3.1.4 Hash Length
+
+The hash length used by the signature. This is the second segment of a [multihash](https://github.com/multiformats/multihash).
+
+### 3.1.5 Content Multicodec Prefix
+
+The [multicodec](https://github.com/multiformats/multicodec) prefix of the encoding scheme used.
+
+For example: `0x00` for raw bytes (no special encoding), `0x0129` for DAG-JSON, and `0x70` for DAG-PB.
+
+### 3.1.6 Raw Signature
+
+The raw signature bytes
+
+## 3.2 Byte Segments
+
+```xml
+<varint 0xD000><varint multicodec_key_prefix><varint multicodec_hash_prefix><varint multicodec_hash_length><varint multicodec_prefix><varint raw_hash><varint raw_signature>
+```
+
+## 3.3 IPLD Schema
+
+``` ipldsch
+type TaggedVarsig union {
+  | Varsig "d0"
+} representation bytesprefix
+
+type Varsig struct {
+  -- Signature type
+  keyPrefix    Multicodec.PublicKey -- Public key type for signature 
+  
+  -- Hash
+  hashPrefix   Multicodec.Multihash -- Hash prefix
+  digestLength Integer
+  
+  -- Payload Encoding
+  contentEnc Multicodec -- Content multicodec prefix
+
+  -- Crypto
+  rawSig     Bytes -- Raw signature bytes
+} representation stringjoin {
+  join ""
```

Comment:
I have _no idea_ if this is how to express this 😅 HALP!

Reply:
Seems reasonable, but not sure if it's legal. 
@warpfork plz halp

Reply:
I think you need to nest byteprefixed unions into each other, but that only works if you have fixed size variants and not going to work for encoding e.g. length. There is related issue here https://github.com/ipld/ipld/issues/241

Reply:
Since IPLD schemas are not actually used for anything other than documentation / specification I suggest we just use fake syntax that conveys idea and is what the linked issue proposes which will end up looking like

```ipldsch
type Varsig struct {
  keyPrefix Multicodec.PublicKey
  hashPrefix Multicodec.Multihash
  digestLength Integer
  contentEnc Multicodec
  rawSig Bytes
} representation multiformat {
  order ["keyPrefix", "hashPrefix", "digestLength", "contentEnc", "rawSig"]
}
```

Reply:
> Since IPLD schemas are not actually used for anything other than documentation / specification

Hmm okay yeah we should probably just cut the fake IPLD Schema syntax, or put a disclaimer on it. Fake schema is probably not helpful versus the `<>` version that's more common in multiformat.